pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -159,32 +189,40 @@ func (d *DualWriterMode2) DeleteCollection(ctx context.Context, deleteValidation
 		return deleted, nil
 	}
 
+	startStorage := time.Now()
 	res, err := d.Storage.DeleteCollection(ctx, deleteValidation, options, &optionsStorage)
 	if err != nil {
 		log.WithValues("deleted", res).Error(err, "failed to delete collection successfully from Storage")
 	}
+	d.recordStorageDuration(err != nil, mode2Str, options.Kind, method, startStorage)
 
 	return res, err
 }
 
 func (d *DualWriterMode2) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
 	log := d.Log.WithValues("name", name, "kind", options.Kind)
 	ctx = klog.NewContext(ctx, log)
+	var method = "delete"
 
+	startLegacy := time.Now()
 	deletedLS, async, err := d.Legacy.Delete(ctx, name, deleteValidation, options)
 	if err != nil {
 		if !apierrors.IsNotFound(err) {
 			log.WithValues("objectList", deletedLS).Error(err, "could not delete from legacy store")
+			d.recordLegacyDuration(true, mode2Str, options.Kind, method, startLegacy)
 			return deletedLS, async, err
 		}
 	}
+	d.recordLegacyDuration(false, mode2Str, options.Kind, method, startLegacy)
 
-	deletedS, _, errUS := d.Storage.Delete(ctx, name, deleteValidation, options)
-	if errUS != nil {
-		if !apierrors.IsNotFound(errUS) {
-			log.WithValues("objectList", deletedS).Error(errUS, "could not delete from duplicate storage")
+	startStorage := time.Now()
+	deletedS, _, err := d.Storage.Delete(ctx, name, deleteValidation, options)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.WithValues("objectList", deletedS).Error(err, "could not delete from duplicate storage")
 		}
 	}
+	d.recordStorageDuration(err != nil, mode2Str, options.Kind, method, startStorage)
```

Comment:
Right now we use this pattern in some places, and 2 separate calls in the cases where we `return` within the if block.  Would it make sense to move this before the if and use the same syntax everywhere?